File: bot.js

const { Telegraf } = require('telegraf');
const axios = require('axios');

// Токен вашего бота
const BOT_TOKEN = '6849461882:AAFgNy2fx3eXzJLJnf1zjwpzpBwuELQCAL8';

// Ваш Telegram ID (используется для отправки сообщений)
const YOUR_TELEGRAM_ID = '1292283830';
// URL API для получения цены EOS (пример для CoinGecko)
const EOS_PRICE_API_URL = 'https://api.coingecko.com/api/v3/simple/price?ids=eos&vs_currencies=usd';
// URL API для получения цены WAXP (пример для CoinGecko)
const WAXP_PRICE_API_URL = 'https://api.coingecko.com/api/v3/simple/price?ids=wax&vs_currencies=usd';
// URL API для получения цены TLOS (пример для CoinGecko)
const TLOS_PRICE_API_URL = 'https://api.coingecko.com/api/v3/simple/price?ids=telos&vs_currencies=usd';

const bot = new Telegraf(BOT_TOKEN);

// Предыдущая цена каждого актива
let prevEOSPrice = null;
let prevWAXPPrice = null;
let prevTLOSPrice = null;

// Функция для получения цены EOS из CoinGecko API
async function getEOSPrice() {
  try {
    const response = await axios.get(EOS_PRICE_API_URL);
    const EOSPriceUSD = response.data.eos.usd;
    return EOSPriceUSD;
  } catch (error) {
    console.error('Ошибка получения цены EOS:', error);
    
  }
}

// Функция для получения цены WAXP из CoinGecko API
async function getWAXPPrice() {
  try {
    const response = await axios.get(WAXP_PRICE_API_URL);
    const WAXPPriceUSD = response.data.wax.usd;
    return WAXPPriceUSD;
  } catch (error) {
    console.error('Ошибка получения цены WAXP:', error);
   
  }
}

// Функция для получения цены TLOS из CoinGecko API
async function getTLOSPrice() {
  try {
    const response = await axios.get(TLOS_PRICE_API_URL);
    const TLOSPriceUSD = response.data.telos.usd;
    return TLOSPriceUSD;
  } catch (error) {
    console.error('Ошибка получения цены TLOS:', error);
   
  }
}

// Функция для отправки сообщения о цене актива
async function sendPriceMessage(coinName, price, percentChange) {
  let emoji = '';
  if (percentChange > 0) {
    emoji = '🟢🚀💹🟢🚀💹';
  } else if (percentChange < 0) {
    emoji = '🔴📉🚨🔴📉🚨';
  }

  const message = `${coinName} ${emoji} Change: ${percentChange.toFixed(2)}%  $${price}`;

  bot.telegram.sendMessage(YOUR_TELEGRAM_ID, message);
}



// Обработчик текстовых сообщений
bot.on('text', async (ctx) => {
  const message = ctx.message.text.toLowerCase();
  if (message.includes('price')) {
    const EOSPrice = await getEOSPrice();
    const WAXPPrice = await getWAXPPrice();
    const TLOSPrice = await getTLOSPrice();
    let response = `Price for EOS: $${EOSPrice}\n`;
    response += `Price for WAXP: $${WAXPPrice}\n`;
    response += `Price for TLOS: $${TLOSPrice}\n`;
    ctx.reply(response);
  }
});

// Функция для отправки сообщений о ценах активов
async function sendPriceMessages() {
  const EOSPrice = await getEOSPrice();
  const WAXPPrice = await getWAXPPrice();
  const TLOSPrice = await getTLOSPrice();

  // Проверка изменения цены для EOS
  if (prevEOSPrice !== null) {
    const EOSPriceChange = EOSPrice - prevEOSPrice;
    const EOSPercentChange = (EOSPriceChange / prevEOSPrice) * 100;
    if (Math.abs(EOSPercentChange) >= 0.1) {
      await sendPriceMessage('EOS', EOSPrice, EOSPercentChange);
    }
  }

  // Проверка изменения цены для WAXP
  if (prevWAXPPrice !== null) {
    const WAXPPriceChange = WAXPPrice - prevWAXPPrice;
    const WAXPPercentChange = (WAXPPriceChange / prevWAXPPrice) * 100;
    if (Math.abs(WAXPPercentChange) >= 0.01) {
      await sendPriceMessage('WAXP', WAXPPrice, WAXPPercentChange);
    }
  }

  // Проверка изменения цены для TLOS
  if (prevTLOSPrice !== null) {
    const TLOSPriceChange = TLOSPrice - prevTLOSPrice;
    const TLOSPercentChange = (TLOSPriceChange / prevTLOSPrice) * 100;
    if (Math.abs(TLOSPercentChange) >= 0.01) {
      await sendPriceMessage('TLOS', TLOSPrice, TLOSPercentChange);
    }
  }

  // Обновление предыдущих цен
  prevEOSPrice = EOSPrice;
  prevWAXPPrice = WAXPPrice;
  prevTLOSPrice = TLOSPrice;
}

// Запуск функции отправки сообщений каждые 20 секунд
setInterval(sendPriceMessages, 30 * 1000);




// Запуск бота
bot.launch();

File: bot.py
import telebot
import requests
import time
import threading

BOT_TOKEN = '6849461882:AAFgNy2fx3eXzJLJnf1zjwpzpBwuELQCAL8'
YOUR_TELEGRAM_ID = '1292283830'

EOS_PRICE_API_URL =  'https://api.binance.com/api/v3/ticker/price?symbol=EOSUSDT'
WAXP_PRICE_API_URL = 'https://api.binance.com/api/v3/ticker/price?symbol=WAXPUSDT'
TLOS_PRICE_API_URL = 'https://api.coingecko.com/api/v3/simple/price?ids=telos&vs_currencies=usd'

prev_EOS_price = None
prev_WAXP_price = None
prev_TLOS_price = None

bot = telebot.TeleBot(BOT_TOKEN)

def get_EOS_price():
    try:
        response = requests.get(EOS_PRICE_API_URL)
        EOS_price_USD = response.json()['price']
        return float(EOS_price_USD)
    except Exception as e:
        print('Ошибка получения цены EOS:', e)
        return None
    
def get_TLOS_price():
    try:
        TLOS_PRICE_API_URL = 'https://api.coingecko.com/api/v3/simple/price?ids=telos&vs_currencies=usd'
        response = requests.get(TLOS_PRICE_API_URL)
        TLOS_price_USD = response.json()['telos']['usd']
        return float(TLOS_price_USD)
    except Exception as e:
        print('Ошибка получения цены TLOS:', e)
        return None

def get_WAXP_price():
    try:
        response = requests.get(WAXP_PRICE_API_URL)
        WAXP_price_USD = response.json()['price']
        return float(WAXP_price_USD)
    except Exception as e:
        print('Ошибка получения цены WAXP:', e)
        return None

def send_price_message(coin_name, price, percent_change):
    emoji = ''
    if percent_change > 0:
        emoji = '🟢🚀💹'
    elif percent_change < 0:
        emoji = '🔴📉🚨'

    message = f'{coin_name} {emoji} Change: {percent_change:.2f}%  ${price}'
    bot.send_message(YOUR_TELEGRAM_ID, message)

def send_price_messages():
    EOS_price = get_EOS_price()
    WAXP_price = get_WAXP_price()
    TLOS_price = get_TLOS_price()

    global prev_EOS_price, prev_WAXP_price, prev_TLOS_price

    if prev_EOS_price is not None:
        EOS_price_change = EOS_price - prev_EOS_price
        EOS_percent_change = (EOS_price_change / prev_EOS_price) * 100
        if abs(EOS_percent_change) >= 0.5:
            send_price_message('EOS', EOS_price, EOS_percent_change)

    if prev_WAXP_price is not None:
        WAXP_price_change = WAXP_price - prev_WAXP_price
        WAXP_percent_change = (WAXP_price_change / prev_WAXP_price) * 100
        if abs(WAXP_percent_change) >= 0.3:
            send_price_message('WAXP', WAXP_price, WAXP_percent_change)

    if prev_TLOS_price is not None:
        TLOS_price_change = TLOS_price - prev_TLOS_price
        TLOS_percent_change = (TLOS_price_change / prev_TLOS_price) * 100
        if abs(TLOS_percent_change) >= 0.3:
            send_price_message('TLOS', TLOS_price, TLOS_percent_change)

    prev_EOS_price = EOS_price
    prev_WAXP_price = WAXP_price
    prev_TLOS_price = TLOS_price

# Обробник команди /start
@bot.message_handler(commands=['start'])
def start_message(message):
    bot.reply_to(message, "GOOD LUCK!🚀")

def get_crypto_prices():
    eos_price = get_EOS_price()
    wax_price = get_WAXP_price()
    telos_price = get_TLOS_price()
    return eos_price, wax_price, telos_price

# Обробник повідомлень, які містять слово "price"
@bot.message_handler(func=lambda message: '/price' in message.text.lower())
def handle_price_message(message):
    # Отримання цін криптовалют
    eos_price, wax_price, telos_price = get_crypto_prices()

    # Створення відповіді з поточними цінами
    response = f"Price NOW:\n"
    response += f"🟢 EOS: 💵{eos_price}\n"
    response += f"🟢 WAXP: 💵{wax_price}\n"
    response += f"🟢 TLOS: 💵{telos_price}\n"



    # Відправлення відповіді користувачеві
    bot.reply_to(message, response)


def bot_polling():
    bot.polling()

# Запускаємо функцію бота в окремому потоці
threading.Thread(target=bot_polling).start()


while True:
    send_price_messages()
    time.sleep(30)



